src/jit/flowgraph.cpp

```diff
@@ -13490,13 +13490,15 @@ bool Compiler::fgOptimizeSwitchBranches(BasicBlock* block)
         GenTree* switchVal = switchTree->gtOp.gtOp1;
         noway_assert(genActualTypeIsIntOrI(switchVal->TypeGet()));
 
+#ifndef LEGACY_BACKEND
```

Comment:
OK I think I see what is going on here.  This node isn't conditionally added, but we "expect" (and assert) that the child node for a switch will be a GT_JMPTABLE (which is only true for the RyuJIT not the LEGACT Jit)